tools/ci/regen_certs.py

```diff
@@ -0,0 +1,53 @@
+import argparse
+import logging
+import subprocess
+
+
+logging.basicConfig()
+logger = logging.getLogger()
+
+
+def get_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--checkend-seconds", type=int, default=5184000,
+                        help="The number of seconds the certificates must be valid for")
+    parser.add_argument("--force", action="store_true",
+                        help="Regenerate certificates even if not reaching expiry")
+    return parser
+
+
+def check_cert(certificate, checkend_seconds):
+    """Checks whether an x509 certificate will expire within a set period.
+
+    Returns 0 if the certificate will not expire, non-zero otherwise."""
+    cmd = [
+        "openssl", "x509",
+        "-checkend", str(checkend_seconds),
+        "-noout",
+        "-in", certificate
+    ]
+    logger.info("Running '%s'" % " ".join(cmd))
+    return subprocess.call(cmd)
+
+
+def regen_certs():
+    """Re-generate the wpt openssl certificates, by delegating to wptserve."""
+    cmd = [
```

Comment:
We could also try to import and use `OpenSSLEnvironment` directly to regenerate the certificates, but it could be prone to breakage since the logic is non-trivial (combine the base config in wptserve with the loaded config json, etc etc).

Reply:
Did you set up `/etc/hosts`? I'm guessing that might be why (Stash trying to bind to e.g. web-platform.tests and get DNS failures.)

Reply:
I don't think that's worthwhile / I'd prefer the current implementation.